### PR TITLE
Bump gatk version to 4.0.4.0.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/gatk/package.py [hunk 1]
```diff
@@ -39,17 +39,26 @@ class Gatk(Package):
 
     version('3.8-0', '0581308d2a25f10d11d3dfd0d6e4d28e', extension='tar.gz',
             url="https://software.broadinstitute.org/gatk/download/auth?package=GATK")
+    version('4.0.4.0', '083d655883fb251e837eb2458141fc2b',
+            url="https://github.com/broadinstitute/gatk/releases/download/4.0.4.0/gatk-4.0.4.0.zip")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gatk/package.py]: Can you sort versions from newest to oldest?

File: var/spack/repos/builtin/packages/gatk/package.py [hunk 2]
```diff
@@ -39,17 +39,26 @@ class Gatk(Package):
...
 
     depends_on('java@8:', type='run')
+    depends_on('python@2.6:2.8,3.6:3.7', type='run', when='@4.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gatk/package.py]: Where did you find these version requirements? I've just never seen something that required exactly Python 3.6 before.

File: var/spack/repos/builtin/packages/gatk/package.py (hunk 1 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/gatk/package.py]: Sure. I'll fix it.

File: var/spack/repos/builtin/packages/gatk/package.py (hunk 2 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/gatk/package.py]: GATK 4 requires "Python 2.6 or greater" or "Python 3.6.2". https://github.com/broadinstitute/gatk Did I specify in the correct way?

File: var/spack/repos/builtin/packages/gatk/package.py [hunk 3]
```diff
@@ -37,19 +37,28 @@ class Gatk(Package):
     url      = "https://software.broadinstitute.org/gatk/download/auth?package=GATK"
     _urlfmt  = "https://github.com/broadgsa/gatk-protected/archive/{0}.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gatk/package.py]: Can you update the homepage to https://software.broadinstitute.org/gatk/. The old link is dead.

Can you also remove the `_urlfmt` and replace the `url` with the new one? Then we might actually be able to find new versions automatically.

**Comment by adamjstewart** [comment]
Flake8:
```
var/spack/repos/builtin/packages/gatk/package.py:54: [E501] line too long (80 > 79 characters)
```

File: var/spack/repos/builtin/packages/gatk/package.py [hunk 4]
```diff
@@ -33,23 +33,31 @@ class Gatk(Package):
     """Genome Analysis Toolkit
        Variant Discovery in High-Throughput Sequencing Data
     """
-    homepage = "http://broadinstitute.github.io/gatk/"
-    url      = "https://software.broadinstitute.org/gatk/download/auth?package=GATK"
-    _urlfmt  = "https://github.com/broadgsa/gatk-protected/archive/{0}.tar.gz"
+    homepage = "https://software.broadinstitute.org/gatk/"
+    url      = "https://github.com/broadinstitute/gatk/releases/download/4.0.4.0/gatk-4.0.4.0.zip")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gatk/package.py]: You need to remove the parenthesis at the end of this line.

**Comment by weijianwen** [comment]
Fixed.

**Pull request merged**